CONTRIBUTING.md

```diff
@@ -1,109 +1,93 @@
-If you are new to GitHub and git, the basic instructions for
```

Comment:
Are you intentionally getting rid of the entire former contents of the CONTRIBUTING.md file? Some parts of it, like the ones about rewrapping and the discouraged file, seem useful for sure. Others are perhaps best kept unless we decide otherwise?

Reply:
I agreee with @jkingdon : This deleted material should be kept. Maybe at/near the end of the page?

Reply:
Sorry, that's my mistake. It was suggested that I move the contribution information to CONTRIBUTING.md, so I did that in a revision to this commit, but in the process I accidentally deleted what was there!

Good news: Since we're using a version control system, it's easy to get that information. I'll try to merge the old/new material into a hopefully-consistent whole.

Reply:
Okay, I've merged the old and new material into something that is hopefully coherent.